arviz/plots/plot_utils.py

```diff
@@ -695,3 +695,25 @@ def _dealiase_sel_kwargs(kwargs, prop_dict, idx):
             {prop: props[idx] for prop, props in prop_dict.items()}, "plot"
         ),
     }
+
+
+def is_valid_quantile(value):
+    """Check if value is a number between 0 and 1."""
+    try:
+        value = float(value)
+        return 0 < value < 1
+    except ValueError:
+        return False
+
+
+def sample_reference_distribution(dist, shape):
+    """Generate samples from a scipy distribution with a given shape."""
+    x_ss = []
+    densities = []
+    dist_rvs = dist.rvs(size=shape)
+    for idx in range(shape[1]):
+        density, xmin, xmax = _fast_kde(dist_rvs[:, idx])
+        x_s = np.linspace(xmin, xmax, len(density))
+        x_ss.append(x_s)
+        densities.append(density)
```

Comment:
we should check memory performance and see if its possible to preallocate. Not sure how it will perform, but specially the append looks suspicious.

Reply:
my memory is foggy but I think that was my first attempt, but then I changed because the size of the grid used by _fast_kde is not fixed and this make cause trouble. That should be fixed with the new upcoming kde. Nevertheless I agree this is something we need to revisit.